core/app/components/SearchDialog.tsx

```diff
@@ -0,0 +1,146 @@
+"use client";
+
+import { useCallback, useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { useQuery } from "@tanstack/react-query";
+import { useDebounce } from "use-debounce";
+
+import {
+	Command,
+	CommandDialog,
+	CommandEmpty,
+	CommandGroup,
+	CommandInput,
+	CommandItem,
+	CommandList,
+} from "ui/command";
+import { AlertCircle, Loader2 } from "ui/icon";
+
+import type { fullTextSearch } from "~/lib/server/pub";
+import { client } from "~/lib/api";
+import { getPubTitle } from "~/lib/pubs";
+import { useCommunity } from "./providers/CommunityProvider";
+
+interface SearchDialogProps {
+	defaultOpen?: boolean;
+	onOpenChange?: (open: boolean) => void;
+	onPubSelect?: (pub: any) => void;
+}
+
+export function SearchDialog({ defaultOpen, onOpenChange, onPubSelect }: SearchDialogProps) {
+	const [open, setOpen] = useState(defaultOpen);
+	const [query, setQuery] = useState("");
+	const [debouncedQuery] = useDebounce(query, 300);
+	const community = useCommunity();
+
+	const {
+		data: results = { status: 400, body: [] },
+		error,
+		isError,
+		isFetching,
+	} = useQuery({
+		queryKey: ["pubs", "search", community.id, debouncedQuery],
+		queryFn: () =>
+			client.pubs.search.query({
+				query: { query: debouncedQuery },
+				params: { communitySlug: community.slug },
+			}),
+		enabled: debouncedQuery.length > 0,
+		placeholderData: (prev, prevQuery) => {
+			if (prevQuery?.state.error || prevQuery?.state.data?.status !== 200) {
+				return prevQuery?.state.data;
+			}
+			return prev;
+		},
+	});
+
+	// CMD+K handler
+	useEffect(() => {
+		const down = (e: KeyboardEvent) => {
+			if (e.key === "k" && (e.metaKey || e.ctrlKey)) {
+				e.preventDefault();
+				onOpenChange?.(true);
+				setOpen(true);
+			}
+		};
+
+		document.addEventListener("keydown", down);
+		return () => document.removeEventListener("keydown", down);
+	}, [onOpenChange]);
```

Comment:
this may be done more elegantly, we should probably have some global thing for this, but for now this is fine i think